src/server/pps/server/pipeline_controller.go

```diff
@@ -455,24 +441,23 @@ func (step *pcStep) setPipelineState(ctx context.Context, state pps.PipelineStat
 // one doesn't exist already), which manages standby and cron inputs, and
 // updates the the pipeline state.
 // Note: this is called by every run through step(), so must be idempotent
-func (step *pcStep) startPipelineMonitor() {
+func (pc *pipelineController) startPipelineMonitor(pi *pps.PipelineInfo) {
 	// since *pc.pipelineInfo may be modified on a Bump, and startMonitor() passes its
 	// input pipelineInfo to a monitor goroutine, send a pointer to a copy of pc.pipelineInfo to
 	// avoid a race condition
-	pi := *step.pipelineInfo
-	if step.pc.monitorCancel == nil {
-		step.pc.monitorCancel = step.pc.startMonitor(step.pc.ctx, &pi)
+	if pc.monitorCancel == nil {
+		// TODO(acohen4): assess whether we should remove &(*pi)
+		pc.monitorCancel = pc.startMonitor(pc.ctx, &(*pi))
```

Comment:
at some point in time we were concerned about this because  `pi` was attached to the `pipelineController`, so we wanted to make sure that monitor goros were working on their own copies. 

I actually just removed this entirely. `pi` is now generated on each `step()` and the monitor goros don't even modify `pi`

Reply:
also good catch on the shallow copy